packages/framework/src/Component/Money/Money.php

```diff
@@ -0,0 +1,213 @@
+        // See https://github.com/Litipk/php-bignumbers/pull/67 for details
+        $decimal = Decimal::fromString((string)$float, $scale);
+
+        return new static($decimal);
+    }
+
+    /**
+     * @param int $integer
+     * @return \Shopsys\FrameworkBundle\Component\Money\Money
+     */
+    public static function fromInteger(int $integer): self
+    {
+        $decimal = Decimal::fromInteger($integer);
+
+        return new static($decimal);
+    }
+
+    /**
+     * @return string
+     */
+    public function toString(): string
+    {
+        return (string)$this->decimal;
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Money\Money $money
+     * @return \Shopsys\FrameworkBundle\Component\Money\Money
+     */
+    public function add(self $money): self
+    {
+        $resultDecimal = $this->decimal->add($money->decimal);
+
+        return new static($resultDecimal);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Money\Money $money
+     * @return \Shopsys\FrameworkBundle\Component\Money\Money
+     */
+    public function subtract(self $money): self
+    {
+        $resultDecimal = $this->decimal->sub($money->decimal);
+
+        return new static($resultDecimal);
+    }
+
+    /**
+     * @param string $multiplier
+     * @return \Shopsys\FrameworkBundle\Component\Money\Money
+     */
+    public function multiply(string $multiplier): self
+    {
+        $decimalMultiplier = Decimal::fromString($multiplier);
+        $resultDecimal = $this->decimal->mul($decimalMultiplier);
+
+        return new static($resultDecimal);
+    }
+
+    /**
+     * @param string $divisor
+     * @param int $scale specifies scale for the division
+     * @return \Shopsys\FrameworkBundle\Component\Money\Money
+     */
+    public function divide(string $divisor, int $scale): self
+    {
+        $decimalDivisor = Decimal::fromString($divisor);
+        $resultDecimal = $this->decimal->div($decimalDivisor, $scale);
+
+        return new static($resultDecimal);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Money\Money $divisor
+     * @return \Shopsys\FrameworkBundle\Component\Money\Money
+     */
+    public function mod(self $divisor): self
+    {
+        $resultDecimal = $this->decimal->mod($divisor->decimal);
+
+        return new static($resultDecimal);
+    }
+
+    /**
+     * @param int $scale
+     * @return \Shopsys\FrameworkBundle\Component\Money\Money
+     */
+    public function round(int $scale): self
+    {
+        $decimal = $this->decimal->round($scale);
+
+        return new static($decimal);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Money\Money $money
+     * @return bool
+     */
+    public function equals(self $money): bool
+    {
+        return $this->decimal->equals($money->decimal);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Money\Money $money
+     * @return int
```

Comment:
We want to keep this method because it is also implemented in MoneyType and Decimal and it is useful for easy sorting in arrays. We have added some info in the annotation

Reply:
:+1: fine reason